packages/xrpl/test/integration/transactions/ammDeposit.test.ts

```diff
@@ -0,0 +1,312 @@
+
+    const { amm: preAmm } = preAmmInfoRes.result
+    const { amount: preAmount, amount2: preAmount2 } = preAmm
+
+    const ammDepositTx: AMMDeposit = {
+      TransactionType: 'AMMDeposit',
+      Account: wallet2.classicAddress,
+      Asset: {
+        currency: 'XRP',
+      },
+      Asset2: {
+        currency: currencyCode,
+        issuer: wallet2.classicAddress,
+      },
+      Amount: '100',
+      Amount2: {
+        currency: currencyCode,
+        issuer: wallet2.classicAddress,
+        value: '100',
+      },
+      Flags: AMMDepositFlags.tfTwoAsset,
+    }
+
+    await testTransaction(testContext.client, ammDepositTx, wallet2)
+
+    const ammInfoRes: AMMInfoResponse = await testContext.client.request({
+      command: 'amm_info',
+      asset: {
+        currency: 'XRP',
+      },
+      asset2: {
+        currency: currencyCode,
+        issuer: wallet2.classicAddress,
+      },
+    })
+
+    const { amm } = ammInfoRes.result
+    const { amount, amount2 } = amm
+
+    if (typeof amount !== 'string') {
+      throw new Error('amount should be a string')
+    }
+    if (typeof preAmount !== 'string') {
+      throw new Error('preAmount should be a string')
+    }
+    assert.equal(parseInt(amount, 10) > parseInt(preAmount, 10), true)
+
+    if (typeof amount2 !== 'object') {
+      throw new Error('amount2 should be an object')
+    }
+    if (typeof preAmount2 !== 'object') {
+      throw new Error('preAmount2 should be an object')
+    }
+    assert.equal(
+      parseInt(amount2.value, 10) > parseInt(preAmount2.value, 10),
+      true,
+    )
+  })
+
+  it('deposit with Amount and LPTokenOut', async function () {
+    const preAmmInfoRes: AMMInfoResponse = await testContext.client.request({
+      command: 'amm_info',
+      asset: {
+        currency: 'XRP',
+      },
+      asset2: {
+        currency: currencyCode,
+        issuer: wallet2.classicAddress,
+      },
+    })
+
+    const { amm: preAmm } = preAmmInfoRes.result
+    const {
+      amount: preAmount,
+      amount2: preAmount2,
+      lp_token: preLPToken,
+    } = preAmm
+
+    const lptokenOut = { ...lptoken, value: '5' }
+    const ammDepositTx: AMMDeposit = {
+      TransactionType: 'AMMDeposit',
+      Account: wallet2.classicAddress,
+      Asset: {
+        currency: 'XRP',
+      },
+      Asset2: {
+        currency: currencyCode,
+        issuer: wallet2.classicAddress,
+      },
+      Amount: '100',
+      LPTokenOut: lptokenOut,
+      Flags: AMMDepositFlags.tfOneAssetLPToken,
+    }
+
+    await testTransaction(testContext.client, ammDepositTx, wallet2)
+
+    const ammInfoRes: AMMInfoResponse = await testContext.client.request({
+      command: 'amm_info',
+      asset: {
+        currency: 'XRP',
+      },
+      asset2: {
+        currency: currencyCode,
+        issuer: wallet2.classicAddress,
+      },
+    })
+
+    const { amm } = ammInfoRes.result
+    const { amount, amount2, lp_token } = amm
+
+    if (typeof amount !== 'string') {
```

Comment:
For these types of tests you can use `expect(amount).toBeInstanceOf(String)`  That will turn these assertions into one line.